talend-component-maven-plugin/src/it/web/.jenkins/scripts/server-check.sh

```diff
@@ -15,31 +15,48 @@
 #  limitations under the License.
 #
 
-set -xe
+# set -xe
 
-main() (
+# Server check
+# Check if the server is running on the giving port
+# Parameters:
+[ -z ${1+x} ] && printf "Parameter 'server_port' use the default value: 8080"
+[ -z ${2+x} ] && printf "Parameter 'server_address' use the default value: http://localhost"
+[ -z ${3+x} ] && printf "Parameter 'timeout' use the default value: 30s"
 
-  serverPort="${1:?Missing server port}"
-  timeout="${2:-30}"
+server_port=${1:-"8080"}
+server_address=${2:-"http://localhost"}
+timeout=${3:-30}
 
-  echo "Waiting server on ${serverPort}..."
+# Check command possibilities
+which curl || { usage "curl is not present"; }
 
-  i=0
+main() (
 
-  which curl || { echo "curl is not present in tsbi image"; exit 1; }
+  echo "Waiting server on ${server_port}"
 
-  while ! curl --output /dev/null --silent --head --fail http://localhost:${serverPort}; do
+  i=0
+
+  while ! curl --output /dev/null --silent --head --fail "${server_address}":"${server_port}"; do
     sleep 1
     ((i = i + 1))
-    echo "."
+    printf "."
```

Comment:
Using printf "." won't return at the line, while echo "." would.
Is this expected ?

Reply:
Yest it is expected